### PR TITLE
CORTX-27708: README docs in Hare should refer to RGW instead of legacy S3

### PR DESCRIPTION
**Pull request by pavankrishnat**
As we moved from legacy S3 to RGW, Hare README documents you gets update.

Solution:
Replaced legacy S3 word with RGW in Hare README documents

Signed-off-by: pavankrishnat <pavan.k.thunuguntla@seagate.com>

### CODE REVIEW

**Comment by SwapnilGaonkar7** [comment]
@pavankrishnat Please include more details in commit message

File: rfc/18/README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ nodes:
           data: [ <str> ]   # e.g. [ "/dev/loop0", "/dev/loop1", "/dev/loop2" ]
                             # Empty list means no IO service.
     m0_clients:
-        s3: <int>     # number of S3 servers to start
+        rgw: <int>     # number of rgw servers to start
```
Comment (SwapnilGaonkar7) [inline rfc/18/README.md:42]: Refer to https://github.com/Seagate/cortx-`hare/blob/main/cfgen/examples/singlenode.yaml` correct format of "m0_clients:"

Comment (pavankrishnat) [reply rfc/18/README.md]: Updated as per reference

**Review by Shreya-18 — APPROVED** [review]

**Review by vaibhavparatwar — APPROVED** [review]

**Comment by vaibhavparatwar** [comment]
@SwapnilGaonkar7  can we merge these changes now? are these looking good?

**Comment by pavankrishnat** [comment]
> @pavankrishnat Please include more details in commit message

Added, Please confirm

File: rfc/20/README.md [hunk 2]
```diff
@@ -146,7 +146,7 @@ Exit codes: 0 if no issues found, 1 otherwise.
 | data_iface      | `cluster>{$server-node}>network>data>interfaces` | This is actually a comma-separated list of strings. The first iface will be taken from that list. |
 | io_disks        | `cluster>{$server-node}>storage>data_devices`    |                                                                                                   |
 | data_iface_type | `cluster>{$server-node}>network>data>interface_type` | Data interface type (tcp|o2ib)                                                                |
-| s3_client_count | `cluster>{$server-node}>s3_instances`            | No of s3 server instances                                                                         |
+| s3_client_count | `cluster>{$server-node}>rgw_instances`            | No of rgw server instances                                                                        |
```
[the remark this answers is not in the document]
Comment (SwapnilGaonkar7) [reply rfc/20/README.md:149]: No, following keys is being used to read motr client info
`cortx>motr>clients`
Above key is list and format for each element is as follows
```
      - name: rgw             
        num_instances: 0              # Optional client. Only for Dev. 
        num_subscriptions: 1
        subscriptions:                # NEW: Optional: Services subscribed for client
        - fdmi        
        num_endpoints: 2
        endpoints:
        - tcp://client-node1:21201
        - tcp://client-node2:21201
```
We use 'num_instances' to get number of instances of client(e.g. rgw)

**Comment by pavankrishnat** [comment]
@SwapnilGaonkar7 @vaibhavparatwar - Addressed review comments, please have a look.

File: README.md [hunk 3]
```diff
@@ -156,7 +156,7 @@ ensure that
 --- | --- | ---
 1 | passwordless `sudo` works for \<user\> | all machines
 2 | \<user\> can `ssh` from \<origin\> to other machines | \<origin\>
-3 | `cortx-hare` and `cortx-s3server` RPMs are installed | all machines
+3 | `cortx-hare` and `cortx-rgw` RPMs are installed | all machines
```
Comment (mssawant) [inline README.md:159]: I would suggest to reword it as, `motr client packages, e.g. cortx-rgw, cortx-s3server, etc.`

File: rfc/11/README.md [hunk 4]
```diff
@@ -20,7 +20,7 @@ Commands:
 
     bootstrap   bootstrap the cluster
     drive-state post drive event to Motr
-    fetch-fids  Fetch the fids for motr services (m0d and s3server)
+    fetch-fids  Fetch the fids for motr services (m0d and rgw)
```
Comment (mssawant) [inline rfc/11/README.md:23]: Same here, let's avoid mentioning rgw explicitly, better mention it as `motr clients, e.g. cortx-rgw, cortx-s3server, etc.`

File: rfc/20/README.md [hunk 5]
```diff
@@ -146,7 +146,8 @@ Exit codes: 0 if no issues found, 1 otherwise.
 | data_iface      | `cluster>{$server-node}>network>data>interfaces` | This is actually a comma-separated list of strings. The first iface will be taken from that list. |
 | io_disks        | `cluster>{$server-node}>storage>data_devices`    |                                                                                                   |
 | data_iface_type | `cluster>{$server-node}>network>data>interface_type` | Data interface type (tcp|o2ib)                                                                |
-| s3_client_count | `cluster>{$server-node}>s3_instances`            | No of s3 server instances                                                                         |
+| name            | `cortx>motr>clients[N]>name`                     | we need to check name of the client is rgw                                                        |
+| instances       | `cortx>motr>clients[N]>num_instances`            | No of rgw server instances, we need to check when name is rgw `(cortx>motr>clients[N]>name)`      |
```
Comment (mssawant) [inline rfc/20/README.md:150]: Same as above in let's avoid explicitly just mentioning rgw, here and at other places in the document.

**Comment by pavankrishnat** [comment]
@mssawant Addressed review comments, please have a look.

File: rfc/6/README.md [hunk 6]
```diff
@@ -76,12 +76,12 @@ Cluster administrator
     - waits for `m0mkfs` to terminate;
     - starts `m0d` process(es).
 
-11. Starts S3 servers.  For each S3 server:
+11. Starts rgw servers.  For each rgw server:
```
Comment (SwapnilGaonkar7) [inline rfc/6/README.md:79]: We do not start RGW or any other client as part of bootstrap now. so lets remove this section altogether

Comment (pavankrishnat) [reply rfc/6/README.md]: Removed

File: rfc/6/README.md [hunk 7]
```diff
@@ -93,4 +87,4 @@ details.
 * Motr ‘confd’ services SHALL be collocated with Consul server agents.
 * Several ‘confd’ service SHALL NOT be running on the same node.
 * Several Motr servers MAY be running on the same node.
-* Several S3 servers MAY be running on the same node.
+* Several rgw servers MAY be running on the same node.
```
Comment (mssawant) [inline rfc/6/README.md:90]: s/rgw/motr clients/ please.

Comment (pavankrishnat) [reply rfc/6/README.md]: Addressed

Comment (pavankrishnat) [reply rfc/6/README.md]: Addressed.

**Review by mssawant — APPROVED** [review]

**Comment by mssawant** [comment]
@pavankrishnat, there are some unresolved conversation from @SwapnilGaonkar7, kindly work it out with him.

**Review by SwapnilGaonkar7 — APPROVED** [review]

**Pull request merged**